src/main/java/io/neonbee/endpoint/metrics/NeonBeePrometheusHandler.java

```diff
@@ -0,0 +1,56 @@
+package io.neonbee.endpoint.metrics;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.composite.CompositeMeterRegistry;
+import io.micrometer.prometheus.PrometheusMeterRegistry;
+import io.neonbee.logging.LoggingFacade;
+import io.prometheus.client.exporter.common.TextFormat;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.micrometer.backends.BackendRegistries;
+import io.vertx.micrometer.impl.PrometheusScrapingHandlerImpl;
+
+/**
+ * A Vert.x Web {@link io.vertx.ext.web.Route} handler for Prometheus metrics scraping.
+ *
+ * The original Implementation doesn't work with {@link CompositeMeterRegistry}. This implementation fixes this.
+ */
+public class NeonBeePrometheusHandler extends PrometheusScrapingHandlerImpl {
+    private static final LoggingFacade LOGGER = LoggingFacade.create();
+
+    private final String registryName;
+
+    /**
+     * Constructs a new instance of NeonBeePrometheusHandler.
+     *
+     * @param registryName The name of the micrometer registry
+     */
+    public NeonBeePrometheusHandler(String registryName) {
+        super(registryName);
+        this.registryName = registryName;
+    }
+
+    @Override
+    public void handle(RoutingContext rc) {
+        MeterRegistry meterRegistry;
+        if (registryName == null) {
+            meterRegistry = BackendRegistries.getDefaultNow();
+        } else {
+            meterRegistry = BackendRegistries.getNow(registryName);
+        }
+
+        if (meterRegistry instanceof CompositeMeterRegistry) {
+            CompositeMeterRegistry cmr = (CompositeMeterRegistry) meterRegistry;
+            cmr.getRegistries().stream().filter(registry -> registry instanceof PrometheusMeterRegistry).findAny()
```

Comment:
We should filter by ID otr name or something like that, to ensure that we ALWAYS get the default PrometheusMeterRegistry. Otherwise we don't know which PrometheusMeterRegistry we get in case that we also use one or multiple custom regestries of type PrometheusMeterRegistry

Reply:
I can't filter the PrometheusMeterRegistry at this point and would therefore suggest that if you configure your own PrometheusMeterRegistry the default PrometheusMeterRegistry in the NeonBeeConfig is not added. If someone really has the case that he has two PrometheusMeterRegistry he would have to solve this himself by writing code.

Reply:
I added the NeonBeePrometheusMeterRegistry class to filter for this registry. I also moved the default initialization of the PrometheusMeterRegistry from the NeonBeeConfig to the MetricsEndpoint. Only when the MetricsEndpoint is loaded, the default NeonBeePrometheusMeterRegistry is added to the CompositeMeterRegistry.

Reply:
Cool! I'm happy that we had a talk.